### DISCUSSION TITLE
Fragments and wildcards

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#1534] [Q&A] [answered]
Hi, is it intentional that langium, unlike Xtext
for fragments does not distinguish between fragments `fragment X` and wildcard fragments `fragment X*`
at all or at least not regarding metamodel inference?

in Xtext
`fragment X` would also produce an interface X and the fragment embedder (`MyRule: something X` would implement it

**Comment by msujew** [comment, marked as answer]
Hey @cdietrich,

Yes, that's by design. At least we didn't necessarily have the need to explicitly generate the types of fragments. I'd be open for a PR that implements those though.

**Reply by spoenemann** [reply]
I think there's not really a need for this in Langium because with TypeScript's type system we can define types the other way around: instead of using interface inheritance everywhere, just create a `type` with a union whenever you need it.
